controllers/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -99,6 +105,200 @@ def test_close(self):
+        self.svc.client.svctask.chvolumegroup.assert_called_once_with(object_id=int(svc_settings.DUMMY_INTERNAL_ID1),
+                                                                      replicationpolicy=REPLICATION_NAME)
+
+    def test_create_ear_replication_not_supported(self):
+        replication_request = ReplicationRequest(OBJECT_INTERNAL_ID, OTHER_OBJECT_INTERNAL_ID, SYSTEM_ID, COPY_TYPE,
+                                                 REPLICATION_TYPE_EAR)
+
+        self.svc.create_replication(replication_request)
+        self.svc.client.svcinfo.mkvolumegroup.assert_not_called()
+        self.svc.client.svcinfo.chvolumegroup.assert_not_called()
+
+    def test_promote_ear_replication_volume_from_independent(self):
+        pass
+
+    def test_promote_ear_replication_volume_from_recovery(self):
+        pass
+
+    def test_promote_ear_replication_not_supported(self):
+        replication = utils.get_mock_mediator_response_replication(name=REPLICATION_NAME,
+                                                                   replication_type=REPLICATION_TYPE_EAR)
+
+        self.svc.promote_replication_volume(replication)
+        self.svc.client.svcinfo.lsvolumegroupreplication.assert_not_called()
+        self.svc.client.svcinfo.chvolumegroupreplication.assert_not_called()
+
+    def test_delete_ear_replication_success(self):
+        self._prepare_mocks_for_ear_replication()
+        replication = utils.get_mock_mediator_response_replication(name=REPLICATION_NAME,
+                                                                   replication_type=REPLICATION_TYPE_EAR,
+                                                                   volume_group_id=OBJECT_INTERNAL_ID)
+        self.svc.delete_replication(replication)
+        self.svc.client.svctask.chvolumegroup.assert_called_once_with(object_id=OBJECT_INTERNAL_ID,
+                                                                      noreplicationpolicy=True)
+        self.svc.client.svctask.rmvolumegroup.assert_called_once_with(object_id=OBJECT_INTERNAL_ID)
+
+    def test_delete_ear_replication_not_supported(self):
+        replication = utils.get_mock_mediator_response_replication(name=REPLICATION_NAME,
+                                                                   replication_type=REPLICATION_TYPE_EAR)
+        self.svc.delete_replication(replication)
+
+        self.svc.client.svcinfo.chvolumegroup.assert_not_called()
+        self.svc.client.svctask.chvdisk.assert_not_called()
+        self.svc.client.svctask.rmvolumegroup.assert_not_called()
+
+    def test_demote_ear_replication_volume(self):
+        self.svc.client.svctask.chvolumereplicationinternals = Mock()
+        replication = utils.get_mock_mediator_response_replication(name=REPLICATION_NAME,
+                                                                   replication_type=REPLICATION_TYPE_EAR)
+        self.svc.demote_replication_volume(replication)
```

Comment:
and..?

Reply:
Duplications moved to function preparing mocks